src/any_agent/frameworks/google.py

```diff
@@ -120,17 +120,17 @@ async def _run_async(  # type: ignore[no-untyped-def]
                         if part.function_response.response.get("success"):
                             final_output = part.function_response.response.get("result")
                             break
-                        if final_output_attempts >= max_output_attepts:
+                        if final_output_attempts >= max_output_attempts:
                             msg = f"Final output failed after {final_output_attempts} attempts"
                             raise ValueError(msg)
 
-                if final_output or final_output_attempts >= max_output_attepts:
+                if final_output or final_output_attempts >= max_output_attempts:
                     break
 
             if not final_output:
                 msg = "No final response found"
                 raise ValueError(msg)
-            return self.config.output_type.model_validate_json(final_output)
+            return self.config.output_type.model_validate(final_output)
```

Comment:
wait, when/where did the `final_output` here changed from string to actual object?

Reply:
This PR changes it from string to a json object (never a pydantic model tho) https://github.com/mozilla-ai/any-agent/pull/626/files#diff-c58e2a0f7f1584ff0bb3fea30ebba3c1dbd0979fef3aec6e586534c0764eb2d0R49